app/components/ui/Footer/Footer.tsx

```diff
@@ -13,6 +13,10 @@ import "./Footer.scss";
 export const Footer = () => {
   const { data, error, isLoading } = useFooterData();
 
+  const res = data as StrapiModel.StrapiDatum<StrapiModel.Footer>;
```

Comment:
It's because `StrapiDataResponse<T>`, which is referenced by the genereic Strapi response type of the fetcher, can be `StrapiDatumResponse<T>`, `Array<StrapiDatumResponse<T>>` or `null`, so to not cast we can do something like `const res: StrapiModel.StrapiDatum<StrapiModel.Footer> | null = data && !Array.isArray(data) ? data : null;`, which may be more clear what's going on to devs.

And if that is all too confusing, we can consider reducing the use of generics. It's a balance of more code and less type checking vs less code and more type checking.